src/admin/components/ResourceTabs/PolygonReviewTab/components/Polygons.tsx

```diff
@@ -92,6 +95,38 @@ const Polygons = (props: IPolygonProps) => {
   } = contextMapArea;
 
   const polygonMenu = useMemo(() => props.menu || [], [props.menu]);
+
+  const filteredPolygons = useMemo(() => {
+    if (!validFilter || validFilter === "all") {
+      return polygonMenu;
+    }
+
+    return polygonMenu.filter(polygon => {
+      const status = polygon.validationStatus;
+
+      if (validFilter === "not_checked") {
+        return !status || status === "" || status === "undefined" || status === "null" || status === "notChecked";
+      }
+
+      return status === validFilter;
+    });
+  }, [polygonMenu, validFilter]);
+
+  const totalPages = Math.max(1, Math.ceil(filteredPolygons.length / pageSize));
+  const startIndex = (currentPage - 1) * pageSize;
+  const endIndex = startIndex + pageSize;
+  const currentPagePolygons = filteredPolygons.slice(startIndex, endIndex);
```

Comment:
This should probably get wrapped in the same useMemo that generates `filteredPolygons`, having that one memo return both.